### PR TITLE
Nice Error Report

### PR DESCRIPTION
**Pull request by winsvega**
if you want additional debugging options
set --verbosity flag for the testeth 

testeth --verbosity 0     // no logs
testeth --verbosity 1     // error reports only  (default)
testeth --verbosity 2     // full detail logs

### CODE REVIEW

File: test/TestHelper.cpp [hunk 1]
```diff
@@ -803,7 +807,27 @@ Options::Options()
...
 		else if (arg == "--fulloutput")
 			fulloutput = true;
+		else if (arg == "--verbosity" && i + 1 < argc)
+		{
+			static std::ostringstream strCout;
+			std::string depthLevel = std::string{argv[i + 1]};
+			if (depthLevel == "0")
+			{
+				logVerbosity = Verbosity::None;
+				std::cout.rdbuf( strCout.rdbuf() );
```
Comment (gavofyork) [inline test/TestHelper.cpp]: style.

File: test/TestHelper.cpp [hunk 2]
```diff
@@ -803,7 +807,27 @@ Options::Options()
...
+				std::cerr.rdbuf( strCout.rdbuf() );
```
Comment (gavofyork) [inline test/TestHelper.cpp]: style

File: test/TestHelper.cpp [hunk 3]
```diff
@@ -803,7 +807,27 @@ Options::Options()
...
+			}
+			else
+			if (depthLevel == "1")
```
Comment (gavofyork) [inline test/TestHelper.cpp]: indent level

**Comment by gavofyork** [comment]
style issues.

**Comment by CJentzsch** [comment]
I did run testeth with the default option with this PR, thats what I got:

./testeth 
Running 821 test cases...
StreamTransformationFilter: invalid PKCS #7 block padding found
CBC: this object cannot use a null IV
StreamTransformationFilter: ciphertext length is not a multiple of block size
StreamTransformationFilter: invalid PKCS #7 block padding found
(shell):1: SyntaxError: Unexpected end of input
[
 
SyntaxError: Unexpected end of input

*** No errors detected

I want at least the name of the test suite and test case that is running, so one can see that it is executing tests. Otherwise you will have no output for a couple of minutes. The output of "ctest" is actually very nice, something similiar would be nice. Also giving a short reoprt at the end over which tests failed.

**Comment by winsvega** [comment]
the default option is for error report only
it does print test suite and test name (I tried to do it where ever I can (BlockChain/StateTests/VMtest at least))
other test suites are quite out of my scope and I suppose I could parse it's errors later. 

if you want more details you could simply run verbosity = 2

**Comment by chriseth** [comment]
I think this is a matter of taste. I prefer a successful test run to output nothing apart from "running X tests...". If a test fails, boost test tells you the name of the test, so you do not need all the clutter from the non-failing tests.

**Comment by winsvega** [comment]
Bosst tells you the name of a test case.  sometime in one test case there are a lot of tests checked. 
In that scenario I added additional output info. 

Do you think this PR is ok to merge?

**Comment by CJentzsch** [comment]
needs a rebase

**Comment by gavofyork** [comment]
builds failing? (maybe rebase to avoid this?)

**Comment by CJentzsch** [comment]
I actually do think this PR breaks the windows build:
 C:\Users\Administrator\buildbot\build-cpp-ethereum-win-pr\build\extdep\install\windows\x64\include\boost/config/requires_threads.hpp(29): fatal error C1189: #error :  "Threading support unavaliable: it has been explicitly disabled with BOOST_DISABLE_THREADS" (boostTest.cpp) [C:\Users\Administrator\buildbot\build-cpp-ethereum-win-pr\build\test\testeth.vcxproj]

**Comment by winsvega** [comment]
windows build errors was due to this line 
#define BOOST_DISABLE_WIN32 //disables SEH warning

File: test/TestHelper.cpp [hunk 4]
```diff
@@ -309,12 +313,13 @@ void ImportTest::compareStates(State const& _stateExpect, State const& _statePos
...
+	return 0;
 }
 
-void ImportTest::exportTest(bytes const& _output)
+int ImportTest::exportTest(bytes const& _output)
 {
-	// export output
-
+	int err = 0;
+	// export output
```
Comment (CJentzsch) [inline test/TestHelper.cpp]: looks like extra spaces or tabs

File: test/TestHelper.cpp [hunk 5]
```diff
@@ -773,7 +787,27 @@ Options::Options()
...
+			static std::ostringstream strCout; //static string to redirect logs to
+			std::string indentLevel = std::string{argv[i + 1]};
+			if (indentLevel == "0")
+			{
+				logVerbosity = Verbosity::None;
+				std::cout.rdbuf(strCout.rdbuf());
+				std::cerr.rdbuf(strCout.rdbuf());
+			}
+			else
+			if (indentLevel == "1")
```
Comment (CJentzsch) [inline test/TestHelper.cpp]: intendation is wrong

File: test/boostTest.cpp [hunk 6]
```diff
@@ -23,6 +23,63 @@
...
+
+		framework::run();
+
+		results_reporter::make_report();
+
+		return runtime_config::no_result_code()
+					? boost::exit_success
+					: results_collector.results( framework::master_test_suite().p_id ).result_code();
+	}
+	catch( framework::nothing_to_test const& ) {
```
Comment (CJentzsch) [inline test/boostTest.cpp]: style:
`catch (framework...)`

File: test/boostTest.cpp [hunk 7]
```diff
@@ -23,6 +23,63 @@
...
+
+	dev::test::Options::get();
+
+	return 0;
+}
+
+//Custom Boost Unit Test Main
+int main( int argc, char* argv[] )
+{
+	try {
```
Comment (CJentzsch) [inline test/boostTest.cpp]: brace always start at the next line

File: test/boostTest.cpp [hunk 8]
```diff
@@ -23,6 +23,63 @@
...
+		framework::init( init_func, argc, argv );
+
+		if( !runtime_config::test_to_run().is_empty() ) {
```
Comment (CJentzsch) [inline test/boostTest.cpp]: same

File: test/boostTest.cpp [hunk 9]
```diff
@@ -23,6 +23,63 @@
...
+
+		framework::run();
+
+		results_reporter::make_report();
+
+		return runtime_config::no_result_code()
+					? boost::exit_success
+					: results_collector.results( framework::master_test_suite().p_id ).result_code();
+	}
+	catch (framework::nothing_to_test const&) {
```
Comment (CJentzsch) [inline test/boostTest.cpp]: same ...

**Pull request merged**